bases with a git repo causes SIGSEGV

Building on the work covered in https://github.com/kubernetes-sigs/kustomize/issues/585, I'm running into the following when doing `$GOPATH/bin/kustomize build`:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x19ecbe0]

goroutine 1 [running]:
sigs.k8s.io/kustomize/pkg/git.(*RepoSpec).Raw(...)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/git/repospec.go:53
sigs.k8s.io/kustomize/pkg/loader.(*fileLoader).errIfPreviouslySeenUri(0xc420275e80, 0xc4200a25a0, 0x60, 0x0, 0xc4202e58f8)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/loader/fileloader.go:226 +0x30
sigs.k8s.io/kustomize/pkg/loader.(*fileLoader).New(0xc420275e80, 0xc4200a25a0, 0x60, 0xc4202e59e0, 0x1011fa9, 0xc4202b5020, 0x20)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/loader/fileloader.go:165 +0x230
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).accumulateBases(0xc420275f80, 0xc4202b5020, 0xc420275f80)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:242 +0x165
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).accumulateTarget(0xc420275f80, 0x0, 0xc4202e5b88, 0x266)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:155 +0x53
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).MakeCustomizedResMap(0xc420275f80, 0xc420275e80, 0x1ce5960, 0x22ffed0)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:116 +0x2f
sigs.k8s.io/kustomize/pkg/commands/build.(*BuildOptions).RunBuild(0xc4202b4720, 0x1cc4a80, 0xc42000e018, 0x1ce5960, 0x22ffed0, 0xc42000e868, 0x1cc8640, 0x22ffed0, 0x0, 0x0)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/commands/build/build.go:115 +0x125
sigs.k8s.io/kustomize/pkg/commands/build.NewCmdBuild.func1(0xc4201c6280, 0x22ffed0, 0x0, 0x0, 0x0, 0x0)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/pkg/commands/build/build.go:78 +0x136
sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra.(*Command).execute(0xc4201c6280, 0x22ffed0, 0x0, 0x0, 0xc4201c6280, 0x22ffed0)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra/command.go:756 +0x468
sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4201c6000, 0xc4201c6780, 0xc4203e4f00, 0xc4203e5400)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra/command.go:846 +0x30a
sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4201c6000, 0xc4201c6000, 0xc4202e5f48)
	/Users/dustin/go/src/sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra/command.go:794 +0x2b
main.main()
	/Users/dustin/go/src/sigs.k8s.io/kustomize/kustomize.go:27 +0x157
```

My kustomization.yaml:

```
bases:
  - git@bitbucket.org:company/project.git//path?ref=branch
```

It looks like support for git@bitbucket.org URLs did work a few days ago. If I checkout 2ef16dc then kustomize works and succesfully pulls from bitbucket.

I can confirm that the 2ef16dc state does fix the issues described in #585 since the latest release of kustomize (v1.0.11 released prior to #585) fails on my kustomization.yaml with:

```
Error: loadResMapFromBasesAndResources: couldn't make ldr for ...
```

**Summary: commits possibly from https://github.com/kubernetes-sigs/kustomize/pull/731 cause a SIGSEGV.**

--- a/pkg/loader/fileloader.go
+++ b/pkg/loader/fileloader.go
@@ -267,7 +267,8 @@ func (l *fileLoader) errIfArgEqualOrHigher(
 // path but a different tag?
 func (l *fileLoader) errIfRepoCycle(newRepoSpec *git.RepoSpec) error {
 	// TODO(monopole): Use parsed data instead of Raw().
-	if strings.HasPrefix(l.repoSpec.Raw(), newRepoSpec.Raw()) {
+	if l.repoSpec != nil &&
+		strings.HasPrefix(l.repoSpec.Raw(), newRepoSpec.Raw()) {
 		return fmt.Errorf(
 			"cycle detected: URI '%s' referenced by previous URI '%s'",
 			newRepoSpec.Raw(), l.repoSpec.Raw())

--- a/pkg/loader/fileloader_test.go
+++ b/pkg/loader/fileloader_test.go
@@ -455,3 +455,29 @@ func TestLoaderDisallowsLocalBaseFromRemoteOverlay(t *testing.T) {
 		t.Fatalf("unexpected err: %v", err)
 	}
 }
+
+func TestLocalLoaderReferencingGitBase(t *testing.T) {
+	topDir := "/whatever"
+	cloneRoot := topDir + "/someClone"
+	fSys := fs.MakeFakeFS()
+	fSys.MkdirAll(topDir)
+	fSys.MkdirAll(cloneRoot + "/foo/base")
+
+	root, err := demandDirectoryRoot(fSys, topDir)
+	if err != nil {
+		t.Fatalf("unexpected err:  %v\n", err)
+	}
+	l1 := newLoaderAtConfirmedDir(
+		root, fSys, nil,
+		git.DoNothingCloner(fs.ConfirmedDir(cloneRoot)))
+	if l1.Root() != topDir {
+		t.Fatalf("unexpected root %s", l1.Root())
+	}
+	l2, err := l1.New("github.com/someOrg/someRepo/foo/base")
+	if err != nil {
+		t.Fatalf("unexpected err:  %v\n", err)
+	}
+	if l2.Root() != cloneRoot+"/foo/base" {
+		t.Fatalf("unexpected root %s", l2.Root())
+	}
+}

